src/sentry/web/frontend/base.py

```diff
@@ -41,83 +34,108 @@
 
 
 class OrganizationMixin:
+    # This attribute will only be set once determine_active_organization is called.  Subclasses should likely invoke
+    # that method, passing along the organization_slug context that might exist (or might not).
+    active_organization: Optional[ApiOrganization]
+
     # TODO(dcramer): move the implicit organization logic into its own class
     # as it's only used in a single location and over complicates the rest of
     # the code
-    def get_active_organization(self, request: Request, organization_slug=None):
-        """
-        Returns the currently active organization for the request or None
-        if no organization.
+    def determine_active_organization(self, request: Request, organization_slug=None) -> NoReturn:
         """
+        Using the current request and potentially optional organization_slug, 'determines'
+        the current session for this mixin object's scope, placing it into the active_organization attribute.
 
-        # TODO(dcramer): this is a huge hack, and we should refactor this
```

Comment:
This "giant hack" existed in the original due to the way this method has side effects, despite being used to "get" the organization in multiple call sites.  My refactoring makes it behave more like middleware -> this method imperatively does the logic and side effects of determining the `active_organization` attribute, and is called by `dispatch` implementations of its subclasses.

Note:  At first, I considered having this base "mixin" do the logic in a base `dispatch` implementation, but actually that was not possible.  Subclasses also have their own custom `dispatch` logic which needs to correctly order the call to `determine_active_organization`.  It's not, great design, but it's only I counted 4 classes, and any other refactoring had much deeper changeset implications.